Fixed request serializer

helloimcarmine opened the pull request:
I've fixed the possibility to use an explicit path during an HTTP request. Moreover I've added a test for this case.

bassosimone commented:
Beware that I cannot even attempt merging because of conflicts with master.

bassosimone commented on example/ndt/ndt:
```diff
@@ -0,0 +1,210 @@
+    $ECHO "See the libtool documentation for more information." 1>&2
+    exit 1
+  fi
+fi
```
You should definitely not commit this file!

bassosimone commented on example/net/discard:
```diff
@@ -0,0 +1,210 @@
+    $ECHO "See the libtool documentation for more information." 1>&2
+    exit 1
+  fi
+fi
```
Ditto

bassosimone commented on example/neubot/dash:
```diff
@@ -0,0 +1,210 @@
+    $ECHO "See the libtool documentation for more information." 1>&2
+    exit 1
+  fi
+fi
```
Ditto!

bassosimone commented:
Uhm, it seems you committed files autogenerated by libtool. I'd remove then first and see whether this improves the state in terms of mergeability.

bassosimone commented:
Uhm, still not possible to merge. Could you make sure that your master branch is up to date and then issue the `git merge master` command in your work branch and fix the possible conflicts at hand? Thank you.

bassosimone commented on src/http/request_serializer.hpp:
```diff
@@ -39,15 +40,32 @@ class RequestSerializer {
         url = parse_url(settings.at("http/url"));
         protocol = settings.get("http/http_version", std::string("HTTP/1.1"));
         method = settings.get("http/method", std::string("GET"));
+>>>>>>> master
```
@helloimcarmine you should manually solve the conflict here by choosing which version of the two you prefer or cherry picking the right lines from the two.

bassosimone commented on src/http/request_serializer.hpp:
```diff
@@ -39,15 +40,26 @@ class RequestSerializer {
+
+        url = parse_url(settings.at("url"));
+        protocol = settings.get("http_version", std::string("HTTP/1.1"));
+        method = settings.get("method", std::string("GET"));
```
This is wrong: you have changed the keys!

bassosimone commented on test/http/request_serializer.cpp:
```diff
@@ -41,3 +41,29 @@ TEST_CASE("HTTP Request serializer works as expected") {
+        {
+         {"follow_redirects", "yes"},
+         {"url", "http://www.example.com/antani?clacsonato=yes#melandri"},
+         {"path", "/antani?clacsonato=no#melandri"}
```
Could you pass something more different so it's easier to spot it?

bassosimone commented:
@helloimcarmine It seems [g++ does not like it when you pass structured default arguments](https://travis-ci.org/measurement-kit/measurement-kit/jobs/135913190#L841). Perhaps this could be fixed by improving `Settings`, but I advise instead to take another route to implement the feature.

bassosimone commented on test/http/request_serializer.cpp:
```diff
@@ -41,3 +41,29 @@ TEST_CASE("HTTP Request serializer works as expected") {
+        {
+         {"http/follow_redirects", "yes"},
+         {"http/url", "http://www.example.com/antani?clacsonato=yes#melandri"},
+         {"http/path", "/antani?amicimiei=yes#melandri"}
```
Here there is a missing colon at end of line

bassosimone commented:
Oh, please disregard my previous comment which was wrong! I've added an inline comment to point out the specific place where the build fails and why.

helloimcarmine commented:
Try now, I've added the missing colon and fixed the expected string, which was wrong. In the previous version a structured default argument was passed too, why is it not accepted now?

bassosimone commented:
> In the previous version a structured default argument was passed too, why is it not accepted now?

See:

>> Oh, please disregard my previous comment which was wrong!

helloimcarmine commented:
Ok sorry, I've understood that it was referred to the missing colon!

bassosimone commented:
No problem!

bassosimone commented:
The diff looks good to me! Merging!
